src/components/PostsList.tsx

```diff
@@ -1,54 +1,65 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-
-import classNames from 'classnames';
-import React from 'react';
+import React, { memo } from 'react';
+import cn from 'classnames';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
 import { Post } from '../types/Post';
+import * as postsActions from '../features/posts';
+import * as commentsActions from '../features/comments';
+
+export const PostsList: React.FC = memo(function PostsListComponent() {
+  const dispatch = useAppDispatch();
+  const { posts, selectedPost, visibleForm } = useAppSelector(state => ({
+    ...state.posts,
+    ...state.comments,
+  }));
 
-type Props = {
-  posts: Post[];
-  selectedPostId?: number;
-  onPostSelected: (post: Post | null) => void;
-};
+  const getComments = (post: Post) => {
+    if (selectedPost?.id === post.id) {
```

Comment:
You can fetch comments inside PostDetails component and get rid of comparing id (just fetch comments for selectedPost)